docs/topics/generics.md

```diff
@@ -328,16 +328,109 @@ At runtime, the instances of generic types do not hold any information about the
 The type information is said to be _erased_. For example, the instances of `Foo<Bar>` and `Foo<Baz?>` are erased to
 just `Foo<*>`.
 
-Therefore, there is no general way to check whether an instance of a generic type was created with certain type
-arguments at runtime, and the compiler [prohibits such `is`-checks](typecasts.md#type-erasure-and-generic-type-checks).
-
-Type casts to generic types with concrete type arguments, for example, `foo as List<String>`, cannot be checked at runtime.
-These [unchecked casts](typecasts.md#unchecked-casts) can be used when type safety is implied by high-level
-program logic but cannot be inferred directly by the compiler. The compiler issues a warning on unchecked casts, and at
-runtime, only the non-generic part is checked (equivalent to `foo as List<*>`).
-
-The type arguments of generic function calls are also only checked at compile time. Inside the function bodies,
-the type parameters cannot be used for type checks, and type casts to type parameters (`foo as T`) are unchecked. However,
-[reified type parameters](inline-functions.md#reified-type-parameters) of inline functions are substituted by the actual
-type arguments in the inlined function body at the call sites and so can be used for type checks and casts,
-with the same restrictions for instances of generic types as described above.
+### Generic type checks and casts
+
+Due to the type erasure, there is no general way to check whether an instance of a generic type was created with certain type
+arguments at runtime, and the compiler prohibits such `is`-checks such as
+`ints is List<Int>` or `list is T` (type parameter). However, you can check an instance against a star-projected type:
+
+```kotlin
+if (something is List<*>) {
+    something.forEach { println(it) } // The items are typed as `Any?`
+}
+```
+
+Similarly, when you already have the type arguments of an instance checked statically (at compile time),
+you can make an `is`-check or a cast that involves the non-generic part of the type. Note that
+angle brackets are omitted in this case:
+
+```kotlin
+fun handleStrings(list: List<String>) {
+    if (list is ArrayList) {
```

Comment:
Funny that this is an example of a [type hole with variance](https://youtrack.jetbrains.com/issue/KT-7972) in Kotlin. It doesn't manifest itself with `String` type argument because it is final, but imagine if `list` was of `List<CharSequence>`: the check for `list is ArrayList` doesn't imply that that array list was created with `CharSequence` argument, therefore casting it to `ArrayList<CharSequence>` would be unsafe.

So I propose at least to declare  list as `MutableList<String>`, so that the example remains valid even when that bug is eventually fixed.

Reply:
Thanks! [Fixed.](https://github.com/JetBrains/kotlin-web-site/pull/2409/commits/ba1930d6ca5035e9d44c667804d9ae473d1553ba)